lessons/jest.md

```diff
@@ -0,0 +1,89 @@
+# ModernJS - Jest
+## Welcome to Modern JS
+
+## Jest
+In this lesson we'll be learning about [Jest](https://facebook.github.io/jest) which is an easy and powerful tool for testing your javascript projects.
+
+### What is Jest?
+It is a unit testing framework, creating with the idea of being a "zero configuration" testing experience.
+We'll spend less time setting up test runners, assertion libraries, mocking tools and more time writing tests, leading to more stable and healthy code bases. 
+
+### Features
+- Automatically generating snapshots for snapshot testing
+- Running tests in parallel, meaning it's super performant
+- Sandboxing tests and automatically tearing down global state so you don't have to worry about conflicts
+- Code coverage reports
+- Built-in mocking and assertions
+
+## Installing Jest
+Jest is available on npm.
+
+```
+// yarn
+yarn add --dev jest
+
+// npm
+npm i --save-dev jest
+```
+
+### Using Jest with babel
+Jest runs on Node and as a result, if you're using ES2016+ you'll need to transpile your code. A nice side effect of this is that we can also write our tests in ES2016+ too.
```

Comment:
Maybe worth mentioning that if you're happy *without* ES modules you can sidestep using Babel if you're using a late Node version as it'll support most, if not all, of the stage-4 stuff. Only if people look intimidated by the babel / jest wiring 😨